validator/client/validator.go

```diff
@@ -938,20 +939,27 @@ func (v *validator) logDuties(slot types.Slot, duties []*ethpb.DutiesResponse_Du
 	}
 	for i := types.Slot(0); i < params.BeaconConfig().SlotsPerEpoch; i++ {
 		startTime := slots.StartTime(v.genesisTime, slotOffset+i)
-		durationTillDuty := time.Until(startTime) + time.Second
+		durationTillDuty := (time.Until(startTime) + time.Second).Round(time.Second)
```

Comment:
Is there a specific reason why we are adding a second here?

Reply:
I think it's so that it rounds up more often than rounds down. It seems a bit strange to me as well. 

I think a better pattern would be to add one second then truncate it. Effectively, that would always round up to the next full second.

```suggestion
		durationTillDuty := (time.Until(startTime) + time.Second).Truncate(time.Second) // Round up to next second.
```